ocaml/typing/printtyp.ml

```diff
@@ -1123,7 +1123,8 @@ let rec tree_of_typexp mode ty =
         Otyp_var (non_gen, Names.name_of_type name_gen tty)
     | Tarrow ((l, marg, mret), ty1, ty2, _) ->
         let lab =
-          if !print_labels || is_optional l then outcome_label l
+          if !print_labels || is_omittable l then outcome_label l
```

Comment:
I really like how we deduced `are_equivalent_with_nolabels` before -- that's what the code was trying to say.

What this code is saying is "is this argument handled specially?". That is, labels just do commuting, and nothing else. So it's reasonable just to omit the labels when printing, if you don't care about labels. But optional and position arguments are more interesting: they change the way the function behaves. So this `is_omittable` is really something like `affects_argument_semantics` or something. But that's a terrible name -- can you come up with anything better? (I can't at the moment.) This new function would have the same implementation as `is_omittable`, but future expansions to the labels system might change this fact.